katnip/targets/pykd_dbg.py

```diff
@@ -0,0 +1,295 @@
+        '''
+        eip = pykd.reg('eip')
+        last_exception = str(pykd.getLastException())
+        exc_code = exceptionInfo.exceptionCode
+        self._Target.logger.info("Got Exception Code: %s at eip:%s" % (hex(exc_code), hex(eip)))
+        if exc_code in interesting_exception_codes.keys():
+            self._Target.is_crash.set()
+            self._Target.crash_dump_finished.clear()
+            self._Target.report.failed("Got Exception Code: %s:%s at eip:%s" % (
+                hex(exc_code), interesting_exception_codes[exc_code], hex(eip)))
+            self._Target.report.add("Error Code", "%s:%s" % (hex(exc_code), interesting_exception_codes[exc_code]))
+            self._Target.report.add("Last Event", "%s" % last_exception)
+            self._Target.report.add("Stacks", str(pykd.dbgCommand("k")))
+            self._Target.crash_dump_finished.set()
+            return pykd.eventResult.Break
+        elif exc_code == break_in_exception_code:
+            # Handle break in event
+            self._Target.logger.info("Break in at eip:%s" % hex(eip))
+            return pykd.eventResult.Break
+        return pykd.eventResult.NoChange
+
+
+class PykdTarget(ServerTarget):
+    '''
+    PykdTarget will run an application for each fuzzed payloads.
+    To use PykdTarget your need install PYkd and Windbg first.
+    Document link https://pykd.codeplex.com/documentation.
+    '''
+
+    def __init__(self, name, process_path, process_args=[], break_points=[], handler=None, logger=None, timeout=3):
+        '''
+
+        :param name: name of the object.
+        :param process_path: path to the target executable.
+        :param process_args: arguments to pass to the process.
+        :param break_points: break points to set.
+        :param handler: pykd event handler.
+        :param logger: logger for this object (default: None)
+        :param timeout: seconds to wait for the process before kill (default: 3)
+
+        :example:
+
+            ::
+                PykdTarget(
+                    name='PykdTarget',
+                    process_path="/tmp/myApp",
+                    process_args=['-a', '-c']
+                    handler=MyEventHandler,
+                    break_points=[],
+                    timeout=2)
+
+            Will run ``/tmp/myApp -a -c mutational_data`` using pykd for evey mutation with timeout of 2 seconds
+
+        '''
+        super(PykdTarget, self).__init__(name, logger)
+        assert(process_path)
+        assert(os.path.exists(process_path))
+        self._process_path = process_path
+        self._process_name = os.path.basename(process_path)
+        self._process_args = process_args
+        self.process_data = None
+        self._break_points = break_points
+        self._handler = handler
+        self.handler = None
+        self._timeout = timeout
+        self._process = None
+        self._pid = None
+        self._system_pid = None
+        self.is_crash = threading.Event()
```

Comment:
all right, I tried to comment each member so you won't forget any, but just fix the relevant ones :smile: